### PR TITLE
Crossword display group clues

### PR DESCRIPTION
**Pull request by NathanielBennett**
For the clues like the mega-clue in http://www.theguardian.com/crosswords/cryptic/26666 ( 2 down ), change the way the clue is displayed so that the order of the word is included ( as shown ). Also, when such a clue is selected highlight all constituent.

clue-nomenclature

clue-hilite

### CODE REVIEW

**Comment by ScottPainterGNM** [comment]
Tiny tiny thing, is it possible to add a space between the final number and the word 'across' (or down in that instance. Just so it looks a bit tidier?

**Comment by ScottPainterGNM** [comment]
This is super awesome otherwise!

**Comment by NathanielBennett** [comment]
I noticed that: It's like that in api: Annoying. http://internal.content.guardianapis.com/crosswords/cryptic/26666?show-fields=all. I'm guessing that this might not always be the case

File: static/src/stylesheets/module/crosswords/_clues.scss [hunk 1]
```diff
@@ -35,3 +35,12 @@
 .crossword__clue--answered {
     color: colour(neutral-2);
 }
+
+.crossword__clue--display-group-order {
+
+    padding-left: 0px;
+
+    &:before {
+        position: relative;
```
Comment (OliverJAsh) [inline static/src/stylesheets/module/crosswords/_clues.scss]: `position: static` is the initial value, or you can just do `position: initial`, if I remember correctly…

I like to leave a comment when I am "undoing CSS", so other people will know.

**Comment by ScottPainterGNM** [comment]
Ah I see, maybe something we can raise with CAPI in that case as would
hopefully be a simple fix

On 8 September 2015 at 16:18, Nathaniel Bennett <notifications@github.com>
wrote:

> I noticed that: It's like that in api: Annoying.
> http://internal.content.guardianapis.com/crosswords/cryptic/26666?show-fields=all.
> I'm guessing that this might not always be the case
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/10433#issuecomment-138597761>.
>



-- 
Scott Painter
QA Analyst | Dotcom

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

File: static/src/javascripts/es6/projects/common/modules/crosswords/clues.js [hunk 2]
```diff
@@ -44,7 +45,8 @@ export default class Clues extends React.Component {
             .map((clue) =>
                 <Clue
                     key={clue.entry.id}
-                    number={clue.entry.number + '.'}
+                    number={clue.entry.number}
+                    displayNumber={clue.entry.humanNumber}
```
Comment (OliverJAsh) [inline static/src/javascripts/es6/projects/common/modules/crosswords/clues.js]: Perhaps we should stay consistent with our terminology here. `humanNumber` works fine?

**Comment by ScottPainterGNM** [comment]
I've emailed Chris Wilk about the CAPI response to see if they can make an amendment

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js [hunk 3]
```diff
@@ -342,13 +342,22 @@ class Crossword extends React.Component {
...
+    }
+
     cluesData () {
         return _.map(this.props.data.entries, (entry) => ({
             entry: entry,
             hasAnswered: _.every(helpers.cellsForEntry(entry), (position) => {
                 return /^[A-Z]$/.test(this.state.grid[position.x][position.y].value);
             }),
-            isSelected: this.clueInFocus() === entry
+            isSelected: this.clueIsInFocusGroup(entry)
```
Comment (OliverJAsh) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: You might be able to re-use `isHighlighted` (in this file), or re-factor it to share code with this. It is responsible for highlighting cells on the grid that are in the current group.

**Comment by OliverJAsh** [comment]
I was just about to raise a PR for this! Oops… oh well.

**Comment by OliverJAsh** [comment]
:+1: !

**Comment by ScottPainterGNM** [comment]
Can this work be extended to factor in the print version under a new PR? It's still missing this data.

**Pull request merged**